distributed/tests/test_worker.py

```diff
@@ -2780,64 +2781,48 @@ async def test_forget_dependents_after_release(c, s, a):
 
 
+@pytest.mark.filterwarnings("ignore:Large object of size")
 @gen_cluster(client=True)
 async def test_steal_during_task_deserialization(c, s, a, b, monkeypatch):
     stealing_ext = s.extensions["stealing"]
     await stealing_ext.stop()
-    from distributed.utils import ThreadPoolExecutor
+    event = Event()
+    event2 = Event()
+    main_thread_id = threading.get_ident()
 
-    class CountingThreadPool(ThreadPoolExecutor):
-        counter = 0
-
-        def submit(self, *args, **kwargs):
-            CountingThreadPool.counter += 1
-            return super().submit(*args, **kwargs)
-
-    # Ensure we're always offloading
-    monkeypatch.setattr("distributed.worker.OFFLOAD_THRESHOLD", 1)
-    threadpool = CountingThreadPool(
-        max_workers=1, thread_name_prefix="Counting-Offload-Threadpool"
-    )
-    try:
-        monkeypatch.setattr("distributed.utils._offload_executor", threadpool)
-
-        class SlowDeserializeCallable:
-            def __init__(self, delay=0.1):
-                self.delay = delay
-
-            def __getstate__(self):
-                return self.delay
-
-            def __setstate__(self, state):
-                delay = state
-                import time
-
-                time.sleep(delay)
-                return SlowDeserializeCallable(delay)
+    class SlowDeserializeCallable:
+        def __init__(self):
+            self.data = b"0" * (OFFLOAD_THRESHOLD + 1)
 
-            def __call__(self, *args, **kwargs):
-                return 41
+        def __getstate__(self):
+            return self.data
 
-        slow_deserialized_func = SlowDeserializeCallable()
-        fut = c.submit(
-            slow_deserialized_func, 1, workers=[a.address], allow_other_workers=True
-        )
+        def __setstate__(self, state):
+            # Ensure this is actually offloaded
+            assert main_thread_id != threading.get_ident()
+            event.set()
+            event2.wait()
+            return SlowDeserializeCallable()
 
-        while CountingThreadPool.counter == 0:
-            await asyncio.sleep(0)
+        def __sizeof__(self):
+            return OFFLOAD_THRESHOLD * 2
 
-        ts = s.tasks[fut.key]
-        a.handle_stimulus(StealRequestEvent(key=fut.key, stimulus_id="test"))
-        stealing_ext.scheduler.send_task_to_worker(b.address, ts)
+        def __call__(self, *args, **kwargs):
+            return 41
 
-        fut2 = c.submit(inc, fut, workers=[a.address])
-        fut3 = c.submit(inc, fut2, workers=[a.address])
+    obj = SlowDeserializeCallable()
+    fut = c.submit(lambda _: 41, obj, workers=[a.address], allow_other_workers=True)
 
-        assert await fut2 == 42
-        await fut3
+    await event.wait()
+    ts = s.tasks[fut.key]
+    a.handle_stimulus(StealRequestEvent(key=fut.key, stimulus_id="test"))
+    stealing_ext.scheduler.send_task_to_worker(b.address, ts)
 
-    finally:
-        threadpool.shutdown()
```

Comment:
somehow, this actually shutdown the actual offload threadpool, not just the mock, i.e. nothing in our test suite was using the offloader threadpool after this test ran 🤯

Reply:
I do not entirely understand why this is shutting down the actual threadpool but I don't care. I removed the mock and it works now